### PR TITLE
EnC: Log more info to in-memory buffer to simplify dump analysis

### PR DESCRIPTION
**Pull request by tmat**

### CODE REVIEW

**Comment by tmat** [comment]
@davidwengier PTAL

File: src/Features/Core/Portable/EditAndContinue/EditSession.cs [hunk 1]
```diff
@@ -380,29 +380,10 @@ private static async ValueTask<bool> HasChangedOrAddedDocumentsAsync(Project old
                    HasChangesThatMayAffectSourceGenerators(oldProject.State, newProject.State);
         }
 
-        private static async Task PopulateChangedAndAddedDocumentsAsync(CommittedSolution oldSolution, Project newProject, ArrayBuilder<Document> changedOrAddedDocuments, CancellationToken cancellationToken)
+        private static async Task PopulateChangedAndAddedDocumentsAsync(Project oldProject, Project newProject, ArrayBuilder<Document> changedOrAddedDocuments, CancellationToken cancellationToken)
         {
             changedOrAddedDocuments.Clear();
 
-            var oldProject = oldSolution.GetProject(newProject.Id);
```
Comment (tmat) [inline src/Features/Core/Portable/EditAndContinue/EditSession.cs:387]: Moved to the caller.

**Review by davidwengier — APPROVED** [review]
This is all still OOP, right? So need a dump of Roslyn.CodeAnalysisService, not devenv?

**Comment by tmat** [comment]
Yup. OOP.

**Pull request merged**